lldb/include/lldb/Target/Target.h

```diff
@@ -627,7 +628,7 @@ class Target : public std::enable_shared_from_this<Target>,
   // used.
   const lldb::ProcessSP &CreateProcess(lldb::ListenerSP listener_sp,
                                        llvm::StringRef plugin_name,
-                                       const FileSpec *crash_file,
+                                       lldb::FileSP crash_file,
                                        bool can_connect);
```

Comment:
I think we can have two interfaces in Target.h: 
- the previous one that taks a "const FileSpec *" 
- the new one that takes a lldb::FileSP
We have a 4 locations that have to manually create a file the code:
```
auto file = FileSystem::Instance().Open(
        filespec, lldb_private::File::eOpenOptionReadOnly);
...
```
This will also help reduce the number of changes in this patch. Then the version that takes a "const FileSpec *" can do the `FileSystem::Instance().Open(...)` and just call the other overload that takes a `lldb::FileSP`

Reply:
If we move the open file code into Target.cpp, it's hard to handle error. Now in these 4 locations we have slightly different ways to report the error when the file is failed to open.
```
        if (!file) {
          result.AppendErrorWithFormatv(
              "Failed to open the core file '{0}': '{1}.'\n",
              core_file.GetPath(), llvm::toString(file.takeError()));
        }
```